### ISSUE TITLE
[feature request] Add support for templates.

### ISSUE THREAD

**Issue by codetalker7** [gsuuon/model.nvim#63]
Apologies in advanced if I'm wrong, but I don't think there's any support for prompt templates (I didn't find anything in the README, atleast)? If not, it'll be really nice to have this. For instance, something even as simple as system/user prompt templates (instead of whole chat templates) will be great.

As a usecase, I want to set the system prompt to something like:

```
system_message = "You are an expert in the {language} programming language. Your job is to assist the user in writing nice programs in {language}".
```

where `{language}` can be filled in.

**Comment by gsuuon** [comment]
The general approach of this plugin right now is programmatic prompt building, so you could write something template-y and `gsub` it. I'd like to do a DX pass at some point to make everything easier to use, and maybe a prompt pack that's actually useful out of the box. If folks are interested getting some template helpers in the built-in utils feel free to drop a thumbs up here as well. Just to double check, you saw the [prompts](https://github.com/gsuuon/model.nvim?tab=readme-ov-file#prompts) section of the readme? It may be a bit too far down now, the docs are due for a rewrite.

**Comment by codetalker7** [comment]
> The general approach of this plugin right now is programmatic prompt building, so you could write something template-y and gsub it.

Yes, that'll be a nice way to do it. Regarding the prompts section; yes, I did take a look. Though I agree, more additions to the documentation will be great! Also, thanks a ton for this amazing plugin!

**Comment by gsuuon** [comment]
Glad you like it!

> Though I agree, more additions to the documentation will be great!

I'd appreciate any specific feedback you have (issue in #28 ), writing docs is probably a weak point of mine.

I'll leave this issue open to track templating utils - would take PR's for this as well.

**Comment by geemus** [comment]
FWIW, when I have a need for something kind of like this I've been using snippets via LuaSnip. It can dump in stuff and jump to the insertion points. For instance, I used this for my current go-to system prompt: https://github.com/geemus/dotfiles/blob/master/lib/config/nvim/LuaSnip/mchat.lua
